services/frontend/adapters/inbound/secret_watcher.go

```diff
@@ -0,0 +1,216 @@
+)
+
+const (
+	watchCreateTimeout = 5 * time.Second
+)
+
+// SecretWatcher uses kubernetes API (Watch method) to scan through secrets in the specified namespace.
+// If the secret contains input manifest label, it is sent through one of the channels in usecases for processing.
+type SecretWatcher struct {
+	// usecases is used to pass changed input manifests.
+	usecases *usecases.Usecases
+	// label is used to identify input manifest secret.
+	label string
+	// namespace is used to identify namespace where secrets are created.
+	namespace string
+	// kubeClient is used to communicate with API server.
+	kubeClient *kubernetes.Clientset
+	// lastResourceVersion is used to resume WATCH API call, after channel is closed.
+	lastResourceVersion string
+}
+
+// NewSecretWatcher returns new SecretWatcher with initialised variables. Returns error if initialisation failed.
+func NewSecretWatcher(usecases *usecases.Usecases) (*SecretWatcher, error) {
+	label, isLabelFound := os.LookupEnv("LABEL")
+	if !isLabelFound {
+		return nil, fmt.Errorf("environment variable LABEL not found")
+	}
+	log.Debug().Msgf("Using LABEL %s", label)
+
+	ns, isNsFound := os.LookupEnv("NAMESPACE")
+	if !isNsFound {
+		return nil, fmt.Errorf("environment variable NAMESPACE not found")
+	}
+	log.Debug().Msgf("Using NAMESPACE %s", ns)
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error while getting in cluster config : %w", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("error while creating client set : %w", err)
+	}
+	secretWatcher := &SecretWatcher{
+		usecases:   usecases,
+		label:      label,
+		kubeClient: clientset,
+		namespace:  ns,
+	}
+
+	return secretWatcher, nil
+}
+
+// Monitor will continuously watch for any changes regarding input manifest secrets. This function will exit once usecases context will get canceled.
+func (sw *SecretWatcher) Monitor() {
+	// Loop the watching as watch.Interface.ResultChan() gets closed after some time.
+	for {
+		select {
+		case <-sw.usecases.Context.Done():
+			// The context is canceled, exit
+			return
+		default:
+			// Create new watcher and monitor.
+			w, err := sw.getNewWatcher()
+			if err != nil {
+				log.Err(err).Msgf("Error while creating WATCH interface, retrying again in %s...", watchCreateTimeout.String())
+				time.Sleep(watchCreateTimeout)
+				continue
+			}
+			// Save lastResourceVersion only if its not empty string
+			if rsv := sw.monitor(w); rsv != "" {
+				sw.lastResourceVersion = rsv
+			}
+		}
+	}
+}
+
+// monitor continuously watches for secrets in current namespace with specified label, until usecases context will be cancelled or watch result channel gets closed.
+func (sw *SecretWatcher) monitor(w watch.Interface) string {
+	rsv := ""
+	for event := range w.ResultChan() {
+		if secret, ok := event.Object.(*v1.Secret); ok {
+			// Save resource version to continue from there once channel gets closed
```

Comment:
The shortest path should be handled first, and having it like this adds unnecessary indentation

can we rewrite this into ?
```go
secret, ok := event.Object.(*v1.Secret)
if !ok {
   continue
}

.....
```